### DISCUSSION TITLE
Plex LXC can't be updated - repository no longer has release file

### DISCUSSION THREAD

**Discussion by Ancalgon** [tteck/Proxmox#2012] [Help Needed]
### Discussed in https://github.com/tteck/Proxmox/discussions/1913

<div type='discussions-op-text'>

<sup>Originally posted by **Ancalgon** October 13, 2023</sup>
Hy guys, 

I need a hand. I have a problem with all my Ubuntu LXC installed via script, PLEX in the first place. The 22.10 verion of ubuntu is not available anymore so I can't update  my Plex setup. I have tried to upgrade repositorys and force an upgrade but every time Plex not working anymore. I'm lucky that Proxmox has a backup system.

Do you have any suggestion?
```
apt update
Hit:1 https://downloads.plex.tv/repo/deb public InRelease
Ign:2 http://archive.ubuntu.com/ubuntu kinetic InRelease
Ign:3 http://archive.ubuntu.com/ubuntu kinetic-updates InRelease
Ign:4 http://archive.ubuntu.com/ubuntu kinetic-security InRelease
Err:5 http://archive.ubuntu.com/ubuntu kinetic Release
  404  Not Found [IP: 185.125.190.36 80]
Err:6 http://archive.ubuntu.com/ubuntu kinetic-updates Release
  404  Not Found [IP: 185.125.190.36 80]
Err:7 http://archive.ubuntu.com/ubuntu kinetic-security Release
  404  Not Found [IP: 185.125.190.36 80]
Reading package lists... Done
E: The repository 'http://archive.ubuntu.com/ubuntu kinetic Release' no longer has a Release file.
N: Updating from such a repository can't be done securely, and is therefore disabled by default.
N: See apt-secure(8) manpage for repository creation and user configuration details.
E: The repository 'http://archive.ubuntu.com/ubuntu kinetic-updates Release' no longer has a Release file.
N: Updating from such a repository can't be done securely, and is therefore disabled by default.
N: See apt-secure(8) manpage for repository creation and user configuration details.
E: The repository 'http://archive.ubuntu.com/ubuntu kinetic-security Release' no longer has a Release file.
N: Updating from such a repository can't be done securely, and is therefore disabled by default.
N: See apt-secure(8) manpage for repository creation and user configuration details.
```
Usign the Plex script inside the container:
`[ERROR] in line 63: exit code 100: while executing command apt-get update &> /dev/null`
</div>

**Comment by X-Anonymous-Y** [comment]
I think the reason is:

Version.          | Code name    | Docs.                | Release.                 | End of Life
Ubuntu 22.10 | Kinetic Kudu | Release Notes | October 20, 2022 | July 20, 2023

Maybe Backup your files and create a New LXC Container with the New Ubuntu Version / LTS Version :)

**Comment by Ancalgon** [comment]
Isn't impossible of course, but I would prefer to continue to use the script in future. And I can't if I'm build my own lxc

**Comment by X-Anonymous-Y** [comment]
Just test the Upgrade command in Terminal:

sudo do-release-upgrade -c

**Comment by Ancalgon** [comment]
I have tried two times but during the upgrade process al the plex related software was removed. And of course the script does not work on a new configured LXC.

**Reply by X-Anonymous-Y** [reply]
does that mean the upgrade itself works? So then all you have to do is install the Plex packages. I think that should work

**Reply by X-Anonymous-Y** [reply]
# Please try this after the upgrade

### Installing dependencies
```BASH
apt-get install -y curl
apt-get install -y sudo
apt-get install -y mc
```

## If it is an "Unprivileged container" run:
### Setting Up Hardware Acceleration
```BASH
apt-get -y install va-driver-all ocl-icd-libopencl1
apt-get install -y intel-opencl-icd

  /bin/chgrp video /dev/dri
  /bin/chmod 755 /dev/dri
  /bin/chmod 660 /dev/dri/*
```

## If is an "Privileged container" run from here and ignore the Above code:
### Setting Up Plex Media Server Repository
```BASH
rm /etc/apt/sources.list.d/plex*
wget -qO- https://downloads.plex.tv/plex-keys/PlexSign.key >/etc/apt/trusted.gpg.d/PlexSign.asc
echo "deb [arch=$(dpkg --print-architecture)] https://downloads.plex.tv/repo/deb/ public main" >/etc/apt/sources.list.d/plexmediaserver.list
```

### Installing Plex Media Server
```BASH
apt-get update
apt-get -o Dpkg::Options::="--force-confold" install -y plexmediaserver
```

### Cleaning up
```BASH
apt-get autoremove
apt-get autoclean
```

# I hope this helps you. If not, I hope that others can help you... Plex is not my thing :) but that's how Plex is currently installed using the script from tteck ;)

# and sorry for my bad english :/

**Reply by X-Anonymous-Y** [reply]
# !!! and run the above commands with root user or with a user with sudo privilegs!

**Reply by X-Anonymous-Y** [reply]
# Another option is maybe to copy data to new LXC

### # Use to copy all data from one Plex Media Server LXC to another
### #  run from the Proxmox Shell

```BASH
bash -c "$(wget -qLO - https://raw.githubusercontent.com/tteck/Proxmox/main/misc/copy-data/plex-copy-data-plex.sh)"
```

**Comment by Ancalgon** [comment]
I've tried to do do upgrade and than followed the commands for my privileged LXC. After the reboot I can clearly see that plex is running

`systemctl status plexmediaserver.service 
* plexmediaserver.service - Plex Media Server
     Loaded: loaded (/lib/systemd/system/plexmediaserver.service; enabled; preset: enabled)
    Drop-In: /etc/systemd/system/plexmediaserver.service.d
             `-override.conf
     Active: active (running) since Thu 2023-11-09 09:50:46 CET; 5min ago
    Process: 470 ExecStartPre=/bin/sh -c /usr/bin/test -d "${PLEX_MEDIA_SERVER_APPLICATION_SUPPORT_DIR}" || /bin/mkdir -p "${PLEX_MEDIA_SERVER_APPLICATION_SUPPORT_DIR}" (cod>
   Main PID: 478 (Plex Media Serv)
      Tasks: 97 (limit: 76672)
     Memory: 206.8M
        CPU: 6.436s
     CGroup: /system.slice/plexmediaserver.service
             |-478 "/usr/lib/plexmediaserver/Plex Media Server"
             |-586 "Plex Plug-in [com.plexapp.system]" /usr/lib/plexmediaserver/Resources/Plug-ins-871adbd44/Framework.bundle/Contents/Resources/Versions/2/Python/bootstrap.>
             |-698 "/usr/lib/plexmediaserver/Plex Tuner Service" /usr/lib/plexmediaserver/Resources/Tuner/Private /usr/lib/plexmediaserver/Resources/Tuner/Shared 1.32.7.7621>
             |-734 "Plex Plug-in [com.plexapp.agents.localmedia]" /usr/lib/plexmediaserver/Resources/Plug-ins-871adbd44/Framework.bundle/Contents/Resources/Versions/2/Python>
             |-736 "Plex Plug-in [com.plexapp.agents.plexthememusic]" /usr/lib/plexmediaserver/Resources/Plug-ins-871adbd44/Framework.bundle/Contents/Resources/Versions/2/Py>
             |-738 "Plex Plug-in [com.plexapp.agents.thetvdb]" /usr/lib/plexmediaserver/Resources/Plug-ins-871adbd44/Framework.bundle/Contents/Resources/Versions/2/Python/bo>
             |-849 "Plex Plug-in [com.plexapp.agents.themoviedb]" /usr/lib/plexmediaserver/Resources/Plug-ins-871adbd44/Framework.bundle/Contents/Resources/Versions/2/Python>
             `-882 "Plex Plug-in [com.plexapp.agents.imdb]" /usr/lib/plexmediaserver/Resources/Plug-ins-871adbd44/Framework.bundle/Contents/Resources/Versions/2/Python/boots>

Nov 09 09:50:46 proxplex systemd[1]: Starting plexmediaserver.service - Plex Media Server...
Nov 09 09:50:46 proxplex systemd[1]: Started plexmediaserver.service - Plex Media Server.
Nov 09 09:50:58 proxplex [698]: Critical: libusb_init failed
Nov 09 09:50:58 proxplex Plex Media Server[698]: Critical: libusb_init failed`

But it's not reachable. If I visit the web UI of my LXC container I get an error from Plex where it's writen that I don't have the authorization to use it. I had tried from my desktop, my smartphon via chrome and via plex app.. nothing, always the same as before
